src/proto/h1/conn.rs

```diff
@@ -61,10 +62,16 @@ where
                 #[cfg(feature = "server")]
                 h1_header_read_timeout: None,
                 #[cfg(feature = "server")]
-                h1_header_read_timeout_fut: None,
-                #[cfg(feature = "server")]
                 h1_header_read_timeout_running: false,
                 #[cfg(feature = "server")]
+                h1_idle_timeout: None,
+                #[cfg(feature = "server")]
+                h1_idle_timeout_running: false,
+                #[cfg(feature = "server")]
+                h1_timeout_fut: None,
+                #[cfg(feature = "server")]
+                first_request: true,
```

Comment:
Side note: It would be \*trivial\* to apply the idle timeout, instead of the header read timeout, to the first request (simply by removing this boolean and treating all requests like `!first_request`).